### PR TITLE
kv/KeyValueDB: add column family

### PR DESCRIPTION
**Pull request by liewegas**
This is another iteration of Adam's squash, but it also cleans up the iterator
mess in KeyValueDB, and simplifies a bunch of the cf handle stuff in rocksdb.

- cleaned up iterators (for KeyValueDB and also the CF one)
- detect existing CFs on open
- add bluestore option to create a CF for deferred and omap

### CODE REVIEW

File: src/kv/KeyValueDB.h [hunk 1]
```diff
@@ -14,18 +14,31 @@
...
+    ColumnFamily(const string &name, const string &option)
+      : name(name), option(option) {}
+  };
+
   class TransactionImpl {
   public:
     /// Set Keys
     void set(
-      const std::string &prefix,                 ///< [in] Prefix for keys
+      const std::string &prefix,                      ///< [in] Prefix for keys, or CF name
```
Comment (amitkumar50) [inline src/kv/KeyValueDB.h]: tab left

**Comment by liewegas** [comment]
@markhpc can you give this one a whirl?  I think it's basically ready to merge.

**Comment by markhpc** [comment]
On it!

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-10-06_15:45:23-rados-wip-sage-testing-2017-10-06-0824-distro-basic-smithi/

**Pull request merged**